examples/toolchain-arm-gcc.cmake

```diff
@@ -0,0 +1,62 @@
+set(CMAKE_SYSTEM_NAME Generic)
+set(CMAKE_SYSTEM_PROCESSOR arm)
+set(CMAKE_TRY_COMPILE_TARGET_TYPE STATIC_LIBRARY)
+
+set(hints_paths)
+
+if (CMAKE_HOST_WIN32)
+    set(X86_PROGRAM_FILES "ProgramFiles(x86)")
+    list(APPEND hints_paths "$ENV{${X86_PROGRAM_FILES}}/GNU Arm Embedded Toolchain/10 2021.10/bin/")
+endif()
+
+find_program(CMAKE_C_COMPILER
+    NAMES arm-none-eabi-gcc
+    HINTS ${hints_paths}
+    REQUIRED)
+
+find_program(CMAKE_CXX_COMPILER
+    NAMES arm-none-eabi-g++
+    HINTS ${hints_paths}
+    REQUIRED)
+
+find_program(CMAKE_ASM_COMPILER
+    NAMES arm-none-eabi-as
+    HINTS ${hints_paths})
+
+find_program(CMAKE_AR
+    NAMES arm-none-eabi-ar
+    HINTS ${hints_paths})
+
+find_program(CMAKE_RANLIB
+    NAMES arm-none-eabi-ranlib
+    HINTS ${hints_paths})
+
+find_program(CMAKE_NM
+    NAMES arm-none-eabi-nm
+    HINTS ${hints_paths})
+
+find_program(EMIL_LD_TOOL
+    NAMES arm-none-eabi-ld
+    HINTS ${hints_paths})
+
+find_program(EMIL_APP_SIZE_TOOL
+    NAMES arm-none-eabi-size
+    HINTS ${hints_paths})
+
+find_program(EMIL_OBJ_COPY_TOOL
+    NAMES arm-none-eabi-objcopy
+    HINTS ${hints_paths})
+
+find_program(EMIL_OBJ_DUMP_TOOL
+    NAMES arm-none-eabi-objdump
+    HINTS ${hints_paths})
+
+find_program(EMIL_DEBUG_TOOL
+    NAMES arm-none-eabi-gdb
+    HINTS ${hints_paths})
+
+add_link_options(LINKER:--gc-sections,--print-memory-usage)
+add_link_options(-specs=nano.specs -nostartfiles -mthumb -mcpu=cortex-m7)
```

Comment:
I would let the top `CMakeLists.txt` define things like `-mthumb` and `-mcpu`

That is not up to the 'generic' arm-gcc toolchain file IMHO

Reply:
I would then rather have architecture-dependent toolchains; I would rather have a clean CMakeLists.txt where details like compiler flags are minimized

Reply:
I agree with Richard here. And since this is only an example I would not be too picky on the flags used. Although we should think about a clean way to handle toolchain files. Duplicating them over all our repositories that need to build one of our HALs is not nice.

Reply:
I totally agree with that as well, its something I've been tinkering with by having a tree structure of supported devices and architectures, where a device file knows which architecture to include